resources/lib/tests/Api/Request/ParametersBuilderTest.php

```diff
@@ -0,0 +1,194 @@
+                        'size' => ['l', 'xl'],
+                        'cat' => 'Category'
+                    ],
+                    Plugin::API_PARAMETER_SORT_ORDER => 'price DESC',
+                    Plugin::API_PARAMETER_PAGINATION_ITEMS_PER_PAGE => 10,
+                    Plugin::API_PARAMETER_PAGINATION_START => 0,
+                ],
+                false,
+                [
+                    'query' => 'Test',
+                    'properties' => [
+                        0 => 'main_variation_id'
+                    ],
+                    'attrib' => [
+                        'size' => ['l', 'xl'],
+                        'cat' => 'Category'
+                    ],
+                    'order' => 'price DESC',
+                    'count' => 10
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider providerSetSearchParams
+     */
+    public function testSetSearchParams($parameters, $category, $expectedParameters)
+    {
+        $requestMock = $this->getMockBuilder(Request::class)->disableOriginalConstructor()->setMethods(null)->getMock();
+
+        $httpRequestMock = $this->getMockBuilder(HttpRequest::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $httpRequestMock->expects($this->once())->method('all')->willReturn($parameters);
+
+        $categoryMock = null;
+        $parametersBuilderMock = $this->getParametersBuilderMock();
+
+        if ($category) {
+            $categoryMock = $this->getMockBuilder(Category::class)->disableOriginalConstructor()->getMock();
+            $categoryMock->parentCategoryId = $category['parentCategoryId'];
+
+            $details = new \stdClass();
+            $details->name = $category['name'];
+
+            $categoryMock->details = [$details];
+        }
+
+        $result = $parametersBuilderMock->setSearchParams($requestMock, $httpRequestMock, $categoryMock);
+
+        $this->assertEquals($expectedParameters, $result->getParams());
+    }
+
+    public function providerGetCategoryName()
+    {
+        return [
+            [
+                [
+                    2 => ['parentCategoryId' => 1, 'name' => 'Test1'],
+                    1 => ['parentCategoryId' => 0, 'name' => 'Test0']
+                ],
+                'Test1_Test0'
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider providerGetCategoryName
+     */
+    public function testGetCategoryName($categories, $expectedCategoryName)
+    {
+        $mockedCategories = [];
+
+        foreach ($categories as $id => $category) {
+            $categoryMock = $this->getMockBuilder(Category::class)->disableOriginalConstructor()->getMock();
+            $categoryMock->parentCategoryId = $category['parentCategoryId'];
+
+            $details = new \stdClass();
+            $details->name = $category['name'];
+            $categoryMock->details = [$details];
+
+            $mockedCategories[] =[$id, $categoryMock];
```

Comment:
You are missing a white space here -> `$mockedCategories[] = [$id, $categoryMock];`